enhancements/network/ovn-k-localnet-api.md

```diff
@@ -0,0 +1,507 @@
+    // Topology describes network configuration.
+    //
+    // Allowed values are "Layer3", "Layer2" and "Localnet".
+    // Layer3 topology creates a layer 2 segment per node, each with a different subnet. Layer 3 routing is used to interconnect node subnets.
+    // Layer2 topology creates one logical switch shared by all nodes.
+    // Localnet topology attach to the overlay local network. Enables egress to the provider's physical network. 
+    //
+    // +kubebuilder:validation:Required
+    // +required
+    // +unionDiscriminator
+    Topology NetworkTopology `json:"topology"`
+    
+    ...
+    
+    // Localnet is the Localnet topology configuration.
+    // +optional
+    Localnet *LocalnetConfig `json:"localnet,omitempty"`
+}
+```
+
+The CUDN spec should have additional validation rule for `spec.network.topology` field:
+```go
+// ClusterUserDefinedNetworkSpec defines the desired state of ClusterUserDefinedNetwork.
+type ClusterUserDefinedNetworkSpec struct {
+    ...
+    // +required
+    Network NetworkSpec `json:"network"`
+}
+```
+
+As of today the CUDN has validation to ensure spec.network is immutable:
+```go
+// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Network spec is immutable"
+```
+This validation should be changed to allow mutation localnet topology configurations, at least MTU and VLAN.
+
+Localnet topology configuration type 
+```go
+type LocalnetConfig struct {
+    // Role describes the network role in the pod.
+    //
+    // Allowed value is "Secondary".
+    // Secondary network is only assigned to pods that use `k8s.v1.cni.cncf.io/networks` annotation to select given network.
+    //
+    // +kubebuilder:validation:Required
+    // +required
+    Role NetworkRole `json:"role"`
+    
+    // PhysicalNetworkName the OVS bridge-mapping's network-name configured in the nodes.
+    //
+    // +kubebuilder:validation:Required
+    // +required
+    PhysicalNetworkName string `json:"name"`
+    
+    // Subnets are used for the pod network across the cluster.
+    // Dual-stack clusters may set 2 subnets (one for each IP family), otherwise only 1 subnet is allowed.
+    //
+    // The format should match standard CIDR notation (for example, "10.128.0.0/16").
+    // This field must be omitted if `ipam.mode` is `Disabled`.
+    //
+    // +optional
+    Subnets DualStackCIDRs `json:"subnets,omitempty"`
+    
+    // ExcludeSubnets List of CIDRs removed from the specified CIDRs in `subnets`.
+    //
+    // The format should match standard CIDR notation (for example, "10.128.0.0/16").
+    // This field must be omitted if `subnetes` is unset or `ipam.mode` is `Disabled`.
+    //
+    // +optional
+    ExcludeSubnets []CIDR `json:"excludeSubnets,omitempty"`
+    
+    // IPAM section contains IPAM-related configuration for the network.
+    //
+    // +optional
+    IPAM *IPAMConfig `json:"ipam,omitempty"`
+    
+    // MTU is the maximum transmission unit for a network.
```

Comment:
```suggestion
    // mtu is the maximum transmission unit for a network.
```

Why does a user care about setting this field? What does it allow them to achieve?